Define-US-for-Data-Consumer-interface

Clementinev opened the pull request:
Define User Stories for [Data Consumer Interface](https://github.com/blindnet-io/product-management/tree/main/refs/high-level-architecture#data-consumer-interface)

See https://github.com/blindnet-io/product-management/issues/756

Close #757

jboileau99 commented on user-stories/US-08-Data-Consumer-interface.md:
```diff
@@ -0,0 +1,113 @@
+
+**because** a submitter might directly ask me where they can submit a Privacy Request 
+
+## US-07-08-12 - other (not part of interface per se)
```
@Clementinev Do we plan to write user stories for the [privacy computation engine](https://github.com/blindnet-io/product-management/tree/main/refs/high-level-architecture#privacy-computation-engine)? Specifically the privacy compiler. If so, would US-07-08-12 and 13 make more sense there?

milstan replied:
IMHO, Privacy Computation Engine (and especially Privacy Compiler) is not a user-facing component and as such is not concerned by user stories. Those compoments aly abide to Functional Requeirements, and other noramtive doduments.

jboileau99 replied:
@milstan I was thinking in our case with building a devkit, the "users" can be developers too, who might be using the privacy computation engine by itself. I see what you mean though, we typically write stories with "end-users" in mind.

milstan replied:
Also, i think our current approach to documentation is by every measure redundant. 

We have high-level/product normative specifications, user stories, issues/functional requirements. And we call ourselves agile :)

The same information is currently present in at lest, more then one place. 

That is another reason why I consider we don't need the user stories for PC.

milstan commented on user-stories/US-08-Data-Consumer-interface.md:
```diff
@@ -0,0 +1,113 @@
+
+## US-07-08-12 - other (not part of interface per se)
+
+**As an** Organization, using blindnet devkit to automate the processing of requests
```
Also there is a need to be able to configure which requests are processed automatically, and which are processed using human validation (Privacy Computation Engine gives a recommendation but a human - Data Consumer/DPO accepts or rjects it).

Clementinev replied:
Add in : 

## US-07-08-14 - other (not part of interface per se)

**As a** developper from an Organization, using blindnet devkit

**I want to** be able to configure which requests are processed automatically, and which are processed using human validation (Privacy Computation Engine gives a recommendation but a human - Data Consumer/DPO accepts or rjects it)

**because** some of the requests can be processed automatically, some others cannot and need human validation

milstan commented:
Few more things to consider here:

**The Privacy Request exchange between Systems might impact the Data Consumer Interface**
- A Privacy Request can arive from another organziation, so I might want to be able to see such requests, where they come from (know that the user did not sumbit them to me, but to someone else who trnasmitted them to me). See [Privacy Request Multicast Protocol draft](https://github.com/blindnet-io/product-management/blob/multicast-protocol/refs/schemas/protocols/RFC-PRMP.md)
- Different Scenarios might apply in this case, I might want to reply to the Data Subject directly, or reply to the System who sent me the Privacy Request (so this system can aswer to the Data Subject). See [Coordinated Reponse](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/scenarios.md#coordinated-response) vs [Direct Response](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/scenarios.md#direct-response)
- When I receive a Privacy Request that has to be shared with a partner organziation I might want to have a setting that each such request is automatically forwarded, or I might want to just get a suggestion to forward it (and to whoom) and explicitely click to forward it.

**A Privacy Request can have multiple demands**
- I might have automatic or semiautomatic responses to some demands and I might need to respond manually to others
- I might deal with some demands today (or they may be dealt with automatically) and with other tomorrow when I have more information of when I get back from vacations.


**I might respond multiple times to the same Privacy Request**
- Nothing should stop me, if I think of something more, to make a second Response to the same Privacy Request I already responded to

**Changes unrelated to Privacy Requests**
- An [Event](https://github.com/blindnet-io/product-management/blob/priv-event/refs/schemas/priv/RFC-PRIV.md#event) can happen, or a [Retention Policy](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/RFC-PRIV.md#retention-policy) might expire, or the user may Revoke Consent (processed automatically) and as a result the Privacy Compiler might emit a recommendation for me to delete particular Fragments. I want to know of such recommendations and react to them.

**Proof of compliance**
- I might want to go back to my actions, look up for a particular Data Subject and see what they asked for, and what I did, and if they ased for data access see if they actually did access the data I gave them. I mgith also want to print that for my boss who is one of those old-fashoned bosses who only has a phone on their desk and a glass of whisky but no computer screen, who reads e-mail after they have been printed and dictates the answers to a secretary.

**I might not be alone in my organisation (not mandatory for Q3)**
- So I might also see what other Data Consumers/DPOs did when looking for proof of compliance
- Also, I need to make sure no other Data Consumer/DPO is processing the same request as I am the same time.

Clementinev commented:
>**The Privacy Request exchange between Systems might impact the Data Consumer Interface**
> * A Privacy Request can arive from another organziation, so I might want to be able to see such requests, where they come from (know that the user did not sumbit them to me, but to someone else who trnasmitted them to me). See [Privacy Request Multicast Protocol draft](https://github.com/blindnet-io/product-management/blob/multicast-protocol/refs/schemas/protocols/RFC-PRMP.md)

add [US-08-19](https://github.com/blindnet-io/product-management/blob/Define-US-for-Data-Consumer-interface/user-stories/US-08-Data-Consumer-interface.md#us-08-19)

> * Different Scenarios might apply in this case, I might want to reply to the Data Subject directly, or reply to the System who sent me the Privacy Request (so this system can aswer to the Data Subject). See [Coordinated Reponse](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/scenarios.md#coordinated-response) vs [Direct Response](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/scenarios.md#direct-response)

add [US-08-20](https://github.com/blindnet-io/product-management/blob/Define-US-for-Data-Consumer-interface/user-stories/US-08-Data-Consumer-interface.md#us-08-20)


> * When I receive a Privacy Request that has to be shared with a partner organziation I might want to have a setting that each such request is automatically forwarded, or I might want to just get a suggestion to forward it (and to whoom) and explicitely click to forward it.

add [US-08-18](https://github.com/blindnet-io/product-management/blob/Define-US-for-Data-Consumer-interface/user-stories/US-08-Data-Consumer-interface.md#us-08-18)

> **A Privacy Request can have multiple demands**
> * I might have automatic or semiautomatic responses to some demands and I might need to respond manually to others

added [US-08-13](https://github.com/blindnet-io/product-management/blob/Define-US-for-Data-Consumer-interface/user-stories/US-08-Data-Consumer-interface.md#us-08-13)


> * I might deal with some demands today (or they may be dealt with automatically) and with other tomorrow when I have more information of when I get back from vacations.

added [US-08-12](https://github.com/blindnet-io/product-management/blob/Define-US-for-Data-Consumer-interface/user-stories/US-08-Data-Consumer-interface.md#us-08-12) 

> **I might respond multiple times to the same Privacy Request**
> * Nothing should stop me, if I think of something more, to make a second Response to the same Privacy Request I already responded to

added [US-08-14](https://github.com/blindnet-io/product-management/blob/Define-US-for-Data-Consumer-interface/user-stories/US-08-Data-Consumer-interface.md#us-08-14)

>**Changes unrelated to Privacy Requests**
> * An [Event](https://github.com/blindnet-io/product-management/blob/priv-event/refs/schemas/priv/RFC-PRIV.md#event) can happen, or a [Retention Policy](https://github.com/blindnet-io/product-management/blob/main/refs/schemas/priv/RFC-PRIV.md#retention-policy) might expire, or the user may Revoke Consent (processed automatically) and as a result the Privacy Compiler might emit a recommendation for me to delete particular Fragments. I want to know of such recommendations and react to them.

added [US-08-15](https://github.com/blindnet-io/product-management/blob/Define-US-for-Data-Consumer-interface/user-stories/US-08-Data-Consumer-interface.md#us-08-15)


>**Proof of compliance**
> * I might want to go back to my actions, look up for a particular Data Subject and see what they asked for, and what I did, and if they ased for data access see if they actually did access the data I gave them. I mgith also want to print that for my boss who is one of those old-fashoned bosses who only has a phone on their desk and a glass of whisky but no computer screen, who reads e-mail after they have been printed and dictates the answers to a secretary.

added [US-08-16](https://github.com/blindnet-io/product-management/blob/Define-US-for-Data-Consumer-interface/user-stories/US-08-Data-Consumer-interface.md#us-08-16), [US-08-17](https://github.com/blindnet-io/product-management/blob/Define-US-for-Data-Consumer-interface/user-stories/US-08-Data-Consumer-interface.md#us-08-17)

milstan commented on user-stories/US-08-Data-Consumer-interface.md:
```diff
@@ -0,0 +1,185 @@
+
+**I want to** process the Privacy requests, that can't be automatically processed, directly from the interface
+
+**because** it is not convenient for me to manually write all an email and manually fetch all elements to answer the Privacy request
```
```suggestion
**because** it is not convenient for me to manually write an email and manually fetch all elements to answer the Privacy request
```

milstan replied:
Here also - I might see a recommended anser, but I might want to modify it, choose a different one.

Clementinev replied:
I updated : 
## US-08-10

**As a** Data Consumer,

**I want to** be guided to answer a Privacy Request, for example having a recommendation that I can modify or a template if I need to add a message to specify why I must deny a Privacy Request

**because** I want to save time and I might not know what can I answer or what are the valid reasons to deny a Privacy request
